packages/schematics/update/update/npm.ts

```diff
@@ -216,17 +224,38 @@ export function getNpmPackageJson(
             token?: string,
             alwaysAuth?: boolean;
             username?: string;
-            password?: string
+            password?: string;
+            email?: string;
           } = {};
 
           if (alwaysAuth !== undefined) {
             auth.alwaysAuth = alwaysAuth === 'false' ? false : !!alwaysAuth;
           }
 
+          if (email) {
+            auth.email = email;
+          }
+
           if (authToken) {
             auth.token = authToken;
           } else if (token) {
-            auth.token = token;
+            try {
+              // attempt to parse "username:password" from base64 token
+              // to enable Artifactory / Nexus-like repositories support
+              const delimiter = ':';
+              const parsedToken = Buffer.from(token, 'base64').toString('ascii');
+              const [extractedUsername, ...passwordArr] = parsedToken.split(delimiter);
+              const extractedPassword = passwordArr.join(delimiter);
+
+              if (extractedUsername && extractedPassword) {
+                auth.username = extractedUsername;
+                auth.password = extractedPassword;
+              } else {
+                throw new Error('Unable to extract username and password from _auth token');
```

Comment:
Why do you try catch and throw an error instead to set auth.token instead of just setting the token in the else block ?

Reply:
because the error message gives more information for the code reader + it is easier later (in case somebody wants it) to throw an error which is way more user-friendly than swallowing it and assigning a token

Reply:
however this is just a matter of taste

Reply:
I get your point, regarding performance it's not really good but we don't realy need performance it that case